Source/OxyPlot/Series/RectangleSeries.cs

```diff
@@ -216,6 +216,11 @@ protected void RenderRectangles(IRenderContext rc, OxyRect clippingRect, ICollec
 
                 var rectrect = new OxyRect(p1, p2);
 
+                if (!clippingRect.Intersects(rectrect))
+                {
+                    continue;
+                }
+
```

Comment:
This isn't really related to the axis margin, or is it? Should this be a separate issue/PR?

Reply:
Yes, it should be a separate PR.